src/common/threading_utils.h

```diff
@@ -129,22 +132,22 @@ class BlockedSpace2d {
   }
 
   std::vector<Range1d> ranges_;
-  std::vector<size_t> first_dimension_;
+  std::vector<std::size_t> first_dimension_;
 };
 
 
 // Wrapper to implement nested parallelism with simple omp parallel for
-template <typename Func>
-void ParallelFor2d(const BlockedSpace2d& space, int nthreads, Func func) {
+inline void ParallelFor2d(BlockedSpace2d const& space, std::int32_t n_threads,
+                          std::function<void(std::size_t, Range1d)> func) {
```

Comment:
Using `std::function` in a tight loop may have a performance implication, since `func` is no longer inlined.

Reply:
This is not used in a tight loop, it's used to create parallel loops.

Reply:
My fault, it's used in a tight loop.

Reply:
But some benchmarks with higgs dataset show no difference before and after the PR. I can revert the change if necessary.

Reply:
You can just revert this line. The constructor can remain the same since it's not in a tight loop.